docs/ERC721.md

```diff
@@ -0,0 +1,686 @@
+        str_to_felt("TKN"),                         # symbol
+        account.contract_address                    # owner
+    ]
+)
+```
+
+As most StarkNet contracts, it expects to be called by another contract and it identifies it through `get_caller_address` (analogous to Solidity's `this.address`). This is why we need an Account contract to interact with it. For example, to mint a non-fungible token:
+
+```python
+signer = Signer(PRIVATE_KEY)
+token_id = uint(1)
+
+account = await starknet.deploy(
+    "contracts/Account.cairo",
+    constructor_calldata=[signer.public_key]
+)
+
+await signer.send_transaction(account, erc721.contract_address, 'mint', [recipient_address, *token_id])
+```
+
+### Token Transfers
+
+EIP721 discourages the use of `transferFrom` and favors `safeTransferFrom` in regard to token transfers; where, the safe function adds the following conditional logic:
+1. if the calling address is an externally owned account (EOA), the token transfer will behave as if `transferFrom` was called
+2. if the calling address is a contract, the safe function will check that the contract supports ERC721 tokens
+
+The current implementation of `safeTansferFrom` checks for `onERC721Received`. 
+Because EOAs cannot be differentiated from regular contracts on StarkNet at this time, it's recommended to transfer tokens with `transferFrom`; however, please be aware that transfering tokens to a contract that does not support ERC721 can result in lost tokens forever. 
+
+### ERC721Received
+
+In order to be sure a contract can safely accept ERC721 tokens, said contract must implement the `ERC721_Receiver` interface. Methods such as `safeTransferFrom` and `safeMint` call the recipient contract's `onERC721Received` method. If the contract fails to return the correct magic value, the transaction fails. 
+
+> Please note: EOA/contract differentiation must be resolved in order to correctly use this feature
+
+### Supporting Interfaces
+
+In order to attain seamless layer one interoperability, this ERC721 implementation uses hardcoded interface IDs that have already been calculated on layer one. The Solidity implementation of ERC165 uses the `super` method to amend parent functions. Being that Cairo/Starknet currently does not support this feature, this implementation uses a mapping to store supported interfaces (similar to OpenZeppelin's [ERC165Storage](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/introspection/ERC165Storage.sol)). 
+
+## Extensibility
+
+This implementation of Cairo smart contract extensibility is set up to include all ERC721 storage and function logic within the base contract `ERC721_base.cairo`. Base contract methods with the prefix `ERC721_` must be imported to the frontend contract and inserted into an `external` method with the requisite name. This is already done in `ERC721.cairo`; however, additional functionality can be added. For instance, you could:
```

Comment:
I think this reads better and it also links to the pattern, otherwise people won't know what extensibility are we talking about.
```suggestion
Following the [contracts extensibility pattern](https://community.starknet.io/t/contract-extensibility-pattern/210), this implementation is set up to include all ERC721 storage and function logic within the base contract `ERC721_base.cairo`. Base contract methods with the prefix `ERC721_` must be imported to the frontend contract and inserted into an `external` method with the requisite name. This is already done in `ERC721.cairo`; however, additional functionality can be added. For instance, you could:
```